hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/qjournal/client/IPCLoggerChannel.java

```diff
@@ -599,7 +599,12 @@ public ListenableFuture<Long> getJournalCTime() {
 
   @Override
   public String toString() {
-    return InetAddresses.toAddrString(addr.getAddress()) + ':' + addr.getPort();
+    if (addr.isUnresolved()) {
+      return addr.getHostName() + ":" + addr.getPort();
```

Comment:
The call to `getHostName()` will force DNS resolution where previously this was not previously forced here. Is this desirable? Problematic? Should be tied to `dfs.namenode.edits.qjournals.resolution-required`?